src/dbus/clientprivate.cpp

```diff
@@ -87,112 +89,40 @@ Ngf::ClientPrivate::~ClientPrivate()
 
 bool Ngf::ClientPrivate::connect()
 {
-    m_connectionWanted = true;
-
-    if (m_iface)
-        return true;
-
     if (!m_serviceWatcher) {
         m_serviceWatcher = new QDBusServiceWatcher(NgfDestination,
                                                    QDBusConnection::systemBus(),
-                                                   QDBusServiceWatcher::WatchForRegistration |
-                                                     QDBusServiceWatcher::WatchForUnregistration,
+                                                   QDBusServiceWatcher::WatchForUnregistration,
                                                    this);
 
-        QObject::connect(m_serviceWatcher, SIGNAL(serviceRegistered(const QString&)),
-                         this, SLOT(serviceRegistered(const QString&)));
         QObject::connect(m_serviceWatcher, SIGNAL(serviceUnregistered(const QString&)),
                          this, SLOT(serviceUnregistered(const QString&)));
 
-        // When connecting to bus for the first time, check whether NGFD exists in the bus.
-        // After we know the initial state it is enough to follow NameOwnerChanged.
-        QDBusMessage msg = QDBusMessage::createMethodCall(QStringLiteral("org.freedesktop.DBus"),
-                                                          QStringLiteral("/org/freedesktop/DBus"),
-                                                          QStringLiteral("org.freedesktop.DBus"),
-                                                          QStringLiteral("GetNameOwner"));
-        QList<QVariant> args;
-        args << NgfDestination;
-        msg.setArguments(args);
-
-        QDBusMessage reply = QDBusConnection::systemBus().call(msg);
-
-        if (reply.type() == QDBusMessage::ErrorMessage) {
-            changeConnected(false);
-            changeAvailable(false);
-            return false;
-        } else {
-            changeAvailable(true);
-        }
-    }
-
-    if (!m_available) {
-        changeConnected(false);
-        return false;
-    }
-
-    QDBusInterface *iface;
-    iface = new QDBusInterface(NgfDestination, NgfPath, NgfInterface,
-                               QDBusConnection::systemBus(), this);
-
-    // Match NGFD signals to get statuses for ongoing events
-    if (iface) {
-        if (iface->isValid()) {
-            iface->connection().connect(QString(), NgfPath, NgfInterface, SignalStatus,
-                                        this, SLOT(setEventState(quint32,quint32)));
-            m_iface = iface;
-            changeConnected(true);
-        } else {
-            iface->deleteLater();
-        }
+        QDBusConnection::systemBus().connect(QString(), NgfPath, NgfInterface, SignalStatus,
+                                             this, SLOT(setEventState(quint32,quint32)));
```

Comment:
nit: should have space after comma?

Reply:
The SIGNAL/SLOT definitions have been a bit special as Qt parses the literal content and it's been seen as optimization to skip them. Here I didn't change anything, rather just moved it a bit.